File: FMS.Sales/FMS.Sales.Domain/Entities/Plan.cs
/*
 * File:          Plan.cs
 * Purpose:       Subscription plan template (Starter / Growth / Pro / Enterprise).
 * Last Modified: 2026-04-29
 */
namespace FMS.Sales.Domain.Entities;

public class Plan
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Code { get; set; } = null!;        // free, starter, growth, pro, enterprise
    public string Name { get; set; } = null!;
    public string? Description { get; set; }
    public int SortOrder { get; set; }
    public bool IsActive { get; set; } = true;
    public bool IsPublic { get; set; } = true;       // false = manual-sales only
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAtUtc { get; set; }

    public ICollection<PlanQuota> Quotas { get; set; } = new List<PlanQuota>();
    public ICollection<PlanFeature> Features { get; set; } = new List<PlanFeature>();
    public ICollection<PlanPrice> Prices { get; set; } = new List<PlanPrice>();
    public ICollection<MeteredPrice> MeteredPrices { get; set; } = new List<MeteredPrice>();
}

File: apps/FMS.Admin/src/types/plan.ts
/**
 * File:          plan.ts
 * Purpose:       Type definitions for operator plan/pricing APIs.
 * Dependencies:  tenant pagination type
 * Last Modified: 2026-05-20
 */
import type { PaginationMetadata } from './tenant';

export type PlanPrice = {
  id: string;
  currencyCode: string;
  billingCycle: string;
  amount: number;
  isActive: boolean;
};

export type PlanQuota = {
  id: string;
  metric: string;
  includedUnits: number;
};

export type PlanFeature = {
  id: string;
  featureKey: string;
  featureValue: string;
};

export type OperatorPlan = {
  id: string;
  code: string;
  name: string;
  description?: string | null;
  sortOrder: number;
  isActive: boolean;
  isPublic: boolean;
  createdAtUtc: string;
  updatedAtUtc?: string | null;
  prices: PlanPrice[];
  quotas: PlanQuota[];
  features: PlanFeature[];
};

export type PlanListPayload = {
  items: OperatorPlan[];
  pagination: PaginationMetadata;
};